### PR TITLE
SSCS-1829 Added cached token generator with config and maven dependency

### PR DESCRIPTION
**Pull request by naveenkvm**

### CODE REVIEW

File: src/main/resources/config/application.properties [hunk 1]
```diff
@@ -26,6 +26,8 @@ notifications.schedule.cron=${NOTIFICATIONS_SCHEDULE_CRON:*/30 * * * * ?}
 max.number.of.notifications.per.run=${MAX_NOTIFICATION_PER_RUN:100}
 max.time.per.run=${MAX_TIME_PER_RUN:60}
 
+auth.provider.service.client=${AUTH_PROVIDER_SERVICE_CLIENT_KEY:test_key}
+
 gov.uk.notification.api.key=${NOTIFICATION_API_KEY:testapikey-cf23bf9c-4a42-42d1-a734-5cfc53dda6ca-edd4df7a-1e31-4d38-a15e-0d55cd93e423}
```
Comment (rkondratowicz) [inline src/main/resources/config/application.properties]: should this be here?

File: src/main/java/uk/gov/hmcts/sscs/service/TokenGenerator.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.auth.provider.service.token.CachedServiceTokenGenerator;
+
+@Service
+public class TokenGenerator {
+	private final CachedServiceTokenGenerator cachedServiceTokenGenerator;
+
+	@Autowired
+	public TokenGenerator(CachedServiceTokenGenerator cachedServiceTokenGenerator){
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/sscs/service/TokenGenerator.java]: is this not used by default?

File: src/main/resources/config/application.properties (hunk 1 above)
Comment (timja-kainos) [inline src/main/resources/config/application.properties]: REMOVE

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,260 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rkondratowicz) [inline pom.xml]: remove this file, add new dependencies to gradle file

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on:
+      service-auth-provider-api:
+        condition: service_started
+    links:
+      - service-auth-provider-api
+
+  service-auth-provider-api:
+    image: docker.artifactory.reform.hmcts.net/auth/service-auth-provider-api:47e18b53aad48ae8124744041988565e07dfed50
+    environment:
+      - auth.provider.service.server.jwtKey=wThK0f0/lh3FlxFcL4xUWDMI5C1J9KyQBgXV4wseh1e5J1uYJIjvTvArHxQDrYoHJ23xFxjHkOnvNbR5dXRoxA==
```
Comment (bartlabno) [inline docker-compose.yml]: I don't think providing any sercets to public is safe. Especially that one

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  tribunals-case-api:
+    build:
+      context: .
+    image: docker.artifactory.reform.hmcts.net/sscs/tribunals-case-api
+    environment:
+      - AUTH_PROVIDER_SERVICE_CLIENT_KEY=JdLxxbOOnulDBbOb
+      - AUTH_PROVIDER_SERVICE_API=http://provider:8080
+      - CCD_SERVICE_API=http://localhost:4451/citizens/%s/jurisdictions/SSCS/case-types/GrantOfRepresentation
+
```
Comment (bartlabno) [inline docker-compose.yml]: Please remove empty lines

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+version: "2.1"
+services:
+
+  tribunals-case-api:
+    build:
+      context: .
+    image: docker.artifactory.reform.hmcts.net/sscs/tribunals-case-api
```
Comment (bartlabno) [inline docker-compose.yml]: is artifatory.reform.hmcts.net available for public? I am not 100% sure.Can you please confirm @timja-kainos ?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    environment:
+      - AUTH_PROVIDER_SERVICE_CLIENT_KEY=JdLxxbOOnulDBbOb
+      - AUTH_PROVIDER_SERVICE_API=http://provider:8080
+      - CCD_SERVICE_API=http://localhost:4451/citizens/%s/jurisdictions/SSCS/case-types/GrantOfRepresentation
+
+    depends_on:
+      service-auth-provider-api:
+        condition: service_started
```
Comment (bartlabno) [inline docker-compose.yml]: Does service_healthy not provide expected results?

File: Dockerfile [hunk 8]
```diff
@@ -2,7 +2,7 @@ FROM openjdk:8-alpine
 
 RUN mkdir -p /opt/sscs/tribunals-case-api
 
-COPY target/tribunals-case-api*.jar /opt/sscs/tribunals-case-api/app.jar
+COPY build/libs/tribunals-case-api*.jar /opt/sscs/tribunals-case-api/app.jar
 
 ENV JAVA_OPTS=""
```
Comment (bartlabno) [inline Dockerfile]: please change `wget -qO- http://localhost:8080/health | grep UP` to `wget -qO- --spider http://localhost:8080/health`

File: docker-compose.yml (hunk 6 above)
Comment (timja-kainos) [reply docker-compose.yml]: no, but its an allowed exception to put here.

File: src/main/java/uk/gov/hmcts/sscs/service/OtpGenerator.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    @SneakyThrows
+    private byte[] calculateHash(long value, byte[] key) throws InvalidKeyException, NoSuchAlgorithmException {
+        SecretKeySpec signKey = new SecretKeySpec(key, "HmacSHA1");
+        Mac mac = Mac.getInstance("HmacSHA1");
+        mac.init(signKey);
+        return mac.doFinal(toBytes(value));
+    }
+
+    private byte[] toBytes(long value) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/sscs/service/OtpGenerator.java]: as you've added Guava:
```
Longs.toByteArray(12345L);
```

**Pull request merged**